### ISSUE TITLE
LOSSAttack and ReferenceAttack implementation

### ISSUE THREAD

**Issue by Happy2Git** [iamgroot42/mimir#31]
Hi, I noticed that the current implementation of LOSSAttack and ReferenceAttack may have an issue with the loss calculation. The `get_ll` method appears to use negative raw probabilities rather than log probabilities, which could affect the accuracy of the attack.

 [[Current Implementation]:](https://github.com/iamgroot42/mimir/blob/90b39b6bec2a3c18c62367f0b9f1daa57a523aa4/mimir/models.py#L147)
```
    @torch.no_grad()
    def get_ll(self,
               text: str,
               tokens: np.ndarray=None,
               probs = None):
        """
            Get the log likelihood of each text under the base_model.

            Args:
                text (str): The input text for which to calculate the log likelihood.
                tokens (numpy.ndarray, optional): An optional array of token ids. If provided, these tokens
                are used instead of tokenizing the input text. Defaults to None.
                probs (list, optional): An optional list of probabilities. If provided, these probabilities
                are used instead of calling the `get_probabilities` method. Defaults to None.
        """
        all_prob = probs if probs is not None else self.get_probabilities(text, tokens=tokens)
        return -np.mean(all_prob)
```

**Comment by XueweiYang209** [comment]
Actually you can find this line
`log_probabilities = torch.nn.functional.log_softmax(shift_logits, dim=-1)`
in the function `Model.get_probabilities()`, so it correctly uses log probabilities
